src/local_neighborhood.py

```diff
@@ -0,0 +1,102 @@
+            node_df.loc[node_df['sources']==True, 'prize'] = 1.0
+            node_df.loc[node_df['targets']==True, 'prize'] = 1.0
+        else:
+            raise ValueError("LocalNeighborhood requires nodes and networks")
+        
+        # input node information
+        node_df.to_csv(filename_map['nodes'],index=False,columns=['NODEID'])
+
+        # For now we assume all input networks are undirected until we expand how edge tables work
+        edges_df = data.get_interactome()
+        edges_df.to_csv(filename_map['network'],sep='|',index=False, columns = ['Interactor1', 'Interactor2'])
+        
+        
+
+
+    @staticmethod
+    def run(network=None, nodes=None, output_file=None, k=None, singularity=False):
+        """
+        Run LocalNeighborhood with Docker
+        @param nodes:  input nodes with sources and targets (required)
+        @param network:  input network file (required)
+        @param output_file: path to the output pathway file (required)
+        @param k: path length (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container
+        """
+        if not nodes or not network or not output_file:
+            raise ValueError('Required LocalNeighborhood arguments are missing')
+
+        work_dir = '/spras'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, node_file = prepare_volume(nodes, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, network_file = prepare_volume(network, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, mapped_output_file = prepare_volume(output_file, work_dir)
+        volumes.append(bind_path)
+
+        command = ['python',
+                   '/LocalNeighborhood/local_neighborhood.py',
+                   '--network', network_file,
+                   '--nodes', node_file,
+                   '--output', mapped_output_file]
+
+        print('Running LocalNeighborhood with arguments: {}'.format(' '.join(command)), flush=True)
+
+        # TODO consider making this a string in the config file instead of a Boolean
+        container_framework = 'singularity' if singularity else 'docker'
+        out = run_container(container_framework,
+                            'zhiqianx/local-neighborhood',
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+
+
+
+    @staticmethod
+    def parse_output(raw_pathway_file, standardized_pathway_file):
+        """
+        Convert a predicted pathway into the universal format
+        @param raw_pathway_file: pathway file produced by an algorithm's run function
+        @param standardized_pathway_file: the same pathway written in the universal format
+        """
+        df = pd.read_csv(raw_pathway_file,sep='|')
+        df['rank'] = 1
+        df.to_csv(standardized_pathway_file, header=False,index=False,sep='\t')
```

Comment:
Inspect the raw-pathway and pathway output files that are produced in the local neighborhood directories. The first row from the raw pathway is being dropped, likely because it is being treated as a header.